### PR TITLE
Remove `IndexTemplateAlreadyExistsException` and `IndexShardAlreadyExistsException`

### PR DESCRIPTION
**Pull request by s1monw**
Both exception can be replaced with java built-in exception, IAE and ISE respectively.
This should be backported partially to 5.x which the transport layer code should be preserved.

Relates to #21494

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/ElasticsearchException.java [hunk 1]
```diff
@@ -553,7 +551,7 @@ public String toString() {
                 org.elasticsearch.common.io.stream.NotSerializableExceptionWrapper::new, 62),
         ALIAS_FILTER_PARSING_EXCEPTION(org.elasticsearch.indices.AliasFilterParsingException.class,
                 org.elasticsearch.indices.AliasFilterParsingException::new, 63),
-        // 64 was DeleteByQueryFailedEngineException, which was removed in 3.0
+        // 64 was DeleteByQueryFailedEngineException, which was removed in 6.0
```
Comment (ywelsch) [inline core/src/main/java/org/elasticsearch/ElasticsearchException.java]: should this be 5.0?

**Review by ywelsch — APPROVED** [review]
left one minor comment. LGTM

File: core/src/main/java/org/elasticsearch/ElasticsearchException.java (hunk 1 above)
Comment (s1monw) [inline core/src/main/java/org/elasticsearch/ElasticsearchException.java]: ah yeah :D I will fix

**Comment by s1monw** [comment]
I also back-ported this to 5.x in d80562dea1da92a46386476cab7a1c6f005f0921

**Pull request merged**